typings/message-bus.d.ts

```diff
@@ -0,0 +1,37 @@
+declare interface MessageBus {
```

Comment:
How do we keep this in sync with both the implementation and the documentation? Is there tooling for checking that, or some way we can work it into our tests? Alternatively, is it possible to include this inline with the existing implementation and parse it out, such that it’s easier to spot when method signatures change but the type definition does not?